dashboard/app/models/user.rb

```diff
@@ -1159,26 +1159,62 @@ def in_progress_and_completed_scripts
     end
   end
 
-  # Return a collection of courses and scripts for the user. First in the list will
-  # be courses enrolled in by the user's sections. Following that will be all scripts
-  # in which the user has made progress that are not in any of the enrolled courses.
-  def recent_courses_and_scripts(exclude_primary_script)
-    courses = section_courses
-    course_scripts_script_ids = courses.map(&:default_course_scripts).flatten.pluck(:script_id).uniq
-
-    # filter out those that are already covered by a course
-    user_scripts = in_progress_and_completed_scripts.
-      select {|user_script| !course_scripts_script_ids.include?(user_script.script_id)}
-
-    course_data = courses.map do |course|
+  def assigned_courses
+    assigned_courses_data = section_courses.map do |course|
       {
         name: course[:name],
         title: data_t_suffix('course.name', course[:name], 'title'),
         description: data_t_suffix('course.name', course[:name], 'description_short'),
         link: course_path(course),
       }
     end
+    assigned_courses_data
+  end
+
+  def assigned_scripts(exclude_hidden)
```

Comment:
Would like to see some function level comments where possible. I realize there's not a great precedent for this in a lot of our Ruby code, but it's something I'd like to see us start getting batter at :)